### ISSUE TITLE
Encrypted password in .ftpconfig

### ISSUE THREAD

**Issue by bitkris-dev** [icetee/remote-ftp#597]
Hello there. I really **love** this package for atom! But there is one critical issue: you can't encrypt your FTP passwords, leaving them in plain text.
Of course I know I can set it to require the password each time, but it is time-consuming when you have **_lots_** of clients and you can't remember the passwords for all of them.

A good solution is the one proposed by the software WinSCP (which is my actual FTP manager, but I would like to leave it, relying only on remote-ftp for atom), which allows you to save passwords for all the websites you want, but encrypting them with a master password. So each time I want to connect to a server I just type my master password, WinSCP encrypts the **true** FTP password and I can connect. (It's a bit like the concept behind password managers)

It could be applied to remote-ftp simply saving the password into the config file as an encrypted string instead of plain text. Unfortunately I don't know how to _technically_ implement a thing like that, I don't know so much about encryption, otherwise I would have done it by myself and sent you a pull request. :/

Hope you find a way for a so brilliant solution.

**Comment by jpxd** [comment]
Its no problem to modify the promptForPassword box to ask for a master password and the encryption would be also not so difficult (although we would need another dependency, which leads to slightly increased package loading times...) but can you suggest a nice and user friendly way to encrypt the password and place it in the .ftpconfig?

**Comment by jpxd** [comment]
What I think of:
We could introduce an "encrypedPassword" field but the password has to get into that somehow. E.g. you place your password one time in the "cleartextPassword" field and at the first launch, you are asked for the master password and the "cleartextPassword" field is encrypted and placed in "cleartextPassword".
The less GUI involved, the faster to implement...

**Comment by bitkris-dev** [comment]
Yes, this is the same solution I thought would be good to implement. If "encrypedPassword" is set to **true**, then during the connection it asks the normal FTP password only for the first time, in order to store it encrypted in the .ftpconfig file.
I don't know how difficult could it be to manipulate the .ftpconfig file itself (you would have to scan all the strings by code and find the position of the value after the parameter, right?), but maybe storing the encrypted password in a different file (e.g.: ".cryptpass") can be easier to manage.
